### PR TITLE
Made 3 arrays into a set

### PR DESCRIPTION
**Pull request by jimmywarting**
**What is the purpose of this pull request? (put an "X" next to item)**

[ ] Documentation update
[ ] Bug fix
[ ] New feature
[x] Other, please explain: Refactor

**What changes did you make? (Give an overview)**
What's looks to be a unik array could instead use Set, and use the delete method instead of 
arrayRemove(item, item.indexOf(onReady))

### CODE REVIEW

**Review by alxhotel — APPROVED** [review]
LGTM

File: lib/conn-pool.js [hunk 1]
```diff
@@ -1,4 +1,3 @@
-const arrayRemove = require('unordered-array-remove')
```
Comment (DiegoRBaquero) [inline lib/conn-pool.js:1]: Could `unordered-array-remove` dep be removed or are the still places where it's used?

**Review by DiegoRBaquero — APPROVED** [review]
LGTM

File: lib/conn-pool.js (hunk 1 above)
Comment (ThaUnknown) [reply lib/conn-pool.js:1]: if you mean this specific file, then it's not used, but unordered-array-remove is still used in other files such as [peer.js](https://github.com/webtorrent/webtorrent/blob/master/lib/peer.js#L272)

Comment (jimmywarting) [reply lib/conn-pool.js:1]: I think i only touched the internals (no public facing properties) that would mean any breaking changes. that's why i think i left out peer.js perhaps.

**Pull request merged**